### PR TITLE
EZP-27920: Repository audit

### PR DESCRIPTION
**Pull request by DominikaK**
First part for https://jira.ez.no/browse/EZP-27920

Also separates permissions.md from repository.md

### CODE REVIEW

File: docs/guide/repository.md [hunk 1]
```diff
@@ -122,234 +122,52 @@ Relations at item level can be of three different types:
...
+When content is moved to a different Location, the item itself and all of its subtree will be assigned to the Section of the new Location. Note that it works only for copy and move; assigning a new section to a parent's Content does not affect the subtree, meaning that Subtree cannot currently be updated this way.
 
-Sections can only be removed if no Content items are assigned to them. Even then, it should be done carefully. When a Section is deleted, it is only its definition itself that will be removed. Other references to the Section will remain and thus the system will most likely be in an inconsistent state.That is why removing Sections may corrupt permission settings, template output and other things in the system.
+Sections can only be removed if no Content items are assigned to them. Even then, it should be done carefully. When a Section is deleted, it is only its definition itself that will be removed. Other references to the Section will remain and thus the system will most likely be in an inconsistent state. That is why removing Sections may corrupt permission settings, template output and other things in the system.
```
Comment (MagdalenaZuba) [inline docs/guide/repository.md]: and the system may be inconsistent.? / become inconsistent?

File: docs/guide/repository.md [hunk 2]
```diff
@@ -432,11 +250,49 @@ This limit is enforced on publishing a new version and only covers archived vers
 
 !!! tip
 
-    Don't set this number too high, with LegacyStorageEngine you'll get performance degradation if you store too many versions. Default value of 5 is in general the recommended value, but the less content you have overall, the more you can increase this to, for instance, 25 or even 50.
+    Don't set this number too high, with Legacy storage engine you'll get performance degradation if you store too many versions. Default value of 5 is in general the recommended value, but the less content you have overall, the more you can increase this to, for instance, 25 or even 50.
```
Comment (MagdalenaZuba) [inline docs/guide/repository.md]: this number -> Repository value

File: docs/guide/repository.md [hunk 3]
```diff
@@ -572,13 +426,13 @@ This will store stash cache to `app/cache/common.`
 
 ##### APC & APCu
```
Comment (MagdalenaZuba) [inline docs/guide/repository.md]: APC & APCu -> APC and APCu

File: docs/guide/repository.md [hunk 4]
```diff
@@ -797,7 +624,126 @@ The Public API exposes Symfony services for all of its repository services.
...
-### Signals Reference
+## SignalSlots
+
+The SignalSlot system provides a means for realizing loosely coupled dependencies in the sense that a code entity A can react on an event occurring in code entity B, without A and B knowing each other directly. This works by dispatching event information through a central third instance, the so called dispatcher:
+
+![Signal Slots diagram](img/signal_slots_diagram.png)
+
+In the shown schematics, object B and one other object are interested in a certain signal. B is a so-called Slot that can be announced to be interested in receiving a Signal (indicated by the circular connector to the dispatcher). Object A now sends the corresponding Signal. The Dispatcher takes care of realizing the dependency and informs the Slot A (and one other Slot) about the occurrence of the Signal.
+
+Signals roughly equal events, while Slots roughly equal event handlers. An arbitrary number (0…n) of Slots can listen for a specific Signal. Every object that receives the Dispatcher as a dependency can send signals. However, the following conditions apply (that typically do not apply to event handling systems):
```
Comment (MagdalenaZuba) [inline docs/guide/repository.md]: signal / Signal - let's use one consistent version - whole doc

File: docs/guide/repository.md [hunk 5]
```diff
@@ -797,7 +624,126 @@ The Public API exposes Symfony services for all of its repository services.
...
+
+Inside the `receive()` method of your Slot you can basically realize any kind of logic. However, it is recommended that you only dispatch the action to be triggered to a dedicated object. This allows you to trigger the same action from within multiple Slots and to re-use the implementation from a completely different context.
+
+Note that, due to the nature of Signal-Slot, the following requirements must be fulfilled by your Slot implementation:
+
+- A Slot must not return anything to the Signal issuer
+- A Slot must be aware that it is potentially executed delayed or even on a different server
+
+**Important**: A single Slot should not take care of processing more than one Signal. Instead, if you need to trigger same or similar actions as different Signals occur. you should encapsulate these actions into a dedicated class, of which your Slots receive an instance to trigger this action.
```
Comment (MagdalenaZuba) [inline docs/guide/repository.md]: **You** should

File: docs/guide/repository.md [hunk 6]
```diff
@@ -797,7 +624,126 @@ The Public API exposes Symfony services for all of its repository services.
...
+As a first step it needs to be determined for which Signals we need to listen in order to update URL aliases. Some of them are:
+
+- `eZ\Publish\Core\SignalSlot\Signal\ContentService\PublishVersionSignal`
+- `eZ\Publish\Core\SignalSlot\Signal\LocationService\CopySubtreeSignal`
+- `eZ\Publish\Core\SignalSlot\Signal\LocationService\MoveSubtreeSignal`
+- ...
+
+There are of course additional Signals that trigger an update of URL aliases, but these are left out for simplicity here.
+
+Now that we identified some Signals to react upon, we can start implementing Slots for these signals. For the first Signal, which is issued as soon as a new version of Content is published, there exists a method in `eZ\Publish\SPI\Persistence\Content\UrlAlias\Handler` for exactly that purpose: `publishUrlAliasForLocation()`. The Signal contains the ID of the content item and its newly published version number. Using this information, the corresponding Slot can fulfill its purposes with the following steps:
```
Comment (MagdalenaZuba) [inline docs/guide/repository.md]: Content item

File: docs/guide/repository.md [hunk 7]
```diff
@@ -756,30 +604,9 @@ $cacheService->clear('content', 'info', $contentId);
...
-
-## Reference
-
-### Services: Public API
+## Services: Public API
 
-The Public API exposes Symfony services for all of its repository services.
+The Public API exposes Symfony services for all of its Repository services.
 
 | Service id                           | Type                                           |
```
Comment (MagdalenaZuba) [inline docs/guide/repository.md]: ID

File: docs/guide/permissions.md [hunk 8]
```diff
@@ -0,0 +1,710 @@
...
+| Value Class     | `eZ\Publish\API\Repository\Values\User\Limitation\ContentTypeLimitation` |
+| Type Class      | `eZ\Publish\Core\Limitation\ContentTypeLimitationType`                   |
+| Criterion used  | `eZ\Publish\API\Repository\Values\Content\Query\Criterion\ContentTypeId` |
+| Role Limitation | no                                                                       |
+
+##### Possible values
+
+|Value|UI value|Description|
+|------|------|------|
+|`<ContentType_id>`|`<ContentType_name>`|All valid ContentType ids can be set as value(s)|
```
Comment (MagdalenaZuba) [inline docs/guide/permissions.md]: IDs or ids = ID or id  - lets use one form for whole doc

File: docs/guide/repository.md [hunk 9]
```diff
@@ -1,54 +1,54 @@
 # Repository
 
-## Introduction
+The content Repository is where all your content is stored.
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/repository.md]: It actually isn't a single file, it's a "thing", or a "place", but there is no good and precise term for this that I can think of.

File: docs/guide/permissions.md [hunk 10]
```diff
@@ -0,0 +1,710 @@
...
+
+|Value|UI value|Description|
+|------|------|------|
+|`1`|"self"|Only the User who is the owner gets access|
+|`2`|"session"|Same as "self"|
+
+##### Legacy compatibility notes:
+
+1. "session" is deprecated and works exactly like "self" in Public API since it has no knowledge of user Sessions
+1. A User is no longer automatically assumed to be the owner of themselves and gets access to edit themselves when the Owner limitation is used in Public API
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/permissions.md]: On second thought I removed this whole section and the similar below it, as they refer to 5.x concerns, so no longer needed

**Comment by DominikaK** [comment]
Also made spelling of "User", "Signal", "ID", "Limitation" consistent.

**Comment by DominikaK** [comment]
I suppose I could also ping @alongosz to take a look :)

**Pull request merged**